File: src/fib.c
#include<stdio.h>
#include<stdlib.h>

int fib(int n);

int fib(int n) {
  if (n <= 1) {
    return n;
  }

  return fib(n - 1) + fib(n - 2);
}


int main(){
  printf("Enter the number to calculate \n");

  int n = getchar();
  printf("%d", fib(n));

  return 0;
}

File: algo/Fib.ts
export default class Fib {
  public cache: number[] = [1];

  public fib(n: number): number {
    if (this.cache[n]) {
      return this.cache[n];
    }
    this.cache[n] = this.fib(n - 1) + this.fib(n - 2);
    return this.cache[n];
  }
}
